aemcmc/gibbs.py

```diff
@@ -455,11 +487,38 @@ def nbinom_dispersion_posterior(
         2012: 1343–50. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4180062/.
 
     """
-    Ls, updates = sample_CRT(srng, y, h_last)
+    h_p = non_constant(h).copy()
+    p_p = non_constant(p).copy()
+    y_p = non_constant(y).copy()
+    a_p = non_constant(a).copy()
+    b_p = non_constant(b).copy()
+
+    Ls, updates = sample_CRT(srng, y_p, h_p)
     L_sum = Ls.sum(axis=-1)
-    h = srng.gamma(a + L_sum, at.reciprocal(b) - at.sum(at.log(1 - p), axis=-1))
-    h.name = f"{h_last.name or 'h'}_posterior"
-    return h, updates
+    h_posterior_p = srng.gamma(
+        a_p + L_sum, at.reciprocal(b_p) - at.sum(at.log(1 - p_p), axis=-1)
+    )
+
+    update_outputs = [h_posterior_p.owner.inputs[0].default_update]
+    update_outputs.extend(updates.values())
+
+    outputs = [h_posterior_p] + update_outputs
+    inputs = [h_p, p_p, a_p, b_p, y_p]
+    gibbs = DispersionGibbsKernel(inputs, outputs, inline=True)
+
+    h_posterior = gibbs(h, p, a, b, y)
+    h_posterior.name = f"{h.name or 'h'}_posterior"
+
+    new_updates = {
+        h_posterior.owner.inputs[-4]: h_posterior.owner.outputs[-1],
+        h_posterior.owner.inputs[-5]: h_posterior.owner.outputs[-2],
+    }
+
+    return h_posterior, new_updates
```

Comment:
@brandonwillard  This change alone breaks `test_create_gibbs`. I suspect there might be an issue with the updates, but I am kind of stuck. Am I missing anything?

Reply:
`h_posterior.owner.inputs` includes _all_ of the shared variables found among all the inputs to the `gibbs` `Op`, so my guess is that entries `-4` and `-5` might not be the correct shared RNG states for the corresponding update values.

Reply:
It was not obvious (to me), but for anyone reading this the relevant shared variables follow the other inputs. here `h_posterior.owner.inputs[5]` and `h_posterior.owner.inputs[6]`.